modules/imgproc/test/test_histograms.cpp

```diff
@@ -1918,5 +1918,28 @@ TEST(Imgproc_Hist_CalcBackProject, accuracy) { CV_CalcBackProjectTest test; test
 TEST(Imgproc_Hist_CalcBackProjectPatch, accuracy) { CV_CalcBackProjectPatchTest test; test.safe_run(); }
 TEST(Imgproc_Hist_BayesianProb, accuracy) { CV_BayesianProbTest test; test.safe_run(); }
 
+TEST(Imgproc_Hist_Calc, calcHist_regression_11544)
+{
+    cv::Mat1w m1 = cv::Mat1w::zeros(10, 10);
+    cv::Mat1w m2 = cv::Mat1w::zeros(10, 10);
+    int n_images = 1;
+    int channels[] = { 0 };
+    cv::Mat mask;
+    cv::MatND hist1, hist2;
+    int dims = 1;
+    int hist_size[] = { 1000 };
+    float range1[] = { 0, 999 };
+    float range2[] = { 0, 1000 };
+    const float* ranges1[] = { range1 };
+    const float* ranges2[] = { range2 };
+
+    cv::calcHist(&m1, n_images, channels, mask, hist1, dims, hist_size, ranges1);
+    cv::calcHist(&m2, n_images, channels, mask, hist2, dims, hist_size, ranges2);
+    for(int i = 0; i < 1000; i++)
+    {
+        EXPECT_EQ(hist1.at<float>(i, 0), hist2.at<float>(i, 0));
```

Comment:
This check looks very strange. It is meaningless checking output similarity for calls with different input parameters.

I believe we should compare "optimized" vs "non-optimized" `calcHist()` calls here:
```
    cv::Mat1w m = cv::Mat1w::zeros(10, 10);
    int n_images = 1;
    int channels[] = { 0 };
    cv::Mat mask;
    cv::Mat hist1, hist2;
    cv::Mat hist1_opt, hist2_opt;
    int dims = 1;
    int hist_size[] = { 1000 };
    float range1[] = { 0, 900 };
    float range2[] = { 0, 1000 };
    const float* ranges1[] = { range1 };
    const float* ranges2[] = { range2 };

    setUseOptimized(false);
    cv::calcHist(&m, n_images, channels, mask, hist1, dims, hist_size, ranges1);
    cv::calcHist(&m, n_images, channels, mask, hist2, dims, hist_size, ranges2);
    setUseOptimized(true);
    cv::calcHist(&m, n_images, channels, mask, hist1_opt, dims, hist_size, ranges1);
    cv::calcHist(&m, n_images, channels, mask, hist2_opt, dims, hist_size, ranges2);
    for(int i = 0; i < 1000; i++)
    {
        EXPECT_EQ(hist1.at<float>(i, 0), hist1_opt.at<float>(i, 0)) << i;
        EXPECT_EQ(hist2.at<float>(i, 0), hist2_opt.at<float>(i, 0)) << i;
    }
```